# Analyzing Used Car Listings on eBay Kleinanzeigen

Dataset was scraped and retrieved from [Kaggle](https://www.kaggle.com/orgesleka/used-cars-database/data)

The data dictionary provided with data is as follows:

-  dateCrawled - When this ad was first crawled. All field-values are taken from this date.
-  name - Name of the car.
-  seller - Whether the seller is private or a dealer.
-  offerType - The type of listing
-  price - The price on the ad to sell the car.
-  abtest - Whether the listing is included in an A/B test.
-  vehicleType - The vehicle Type.
-  yearOfRegistration - The year in which which year the car was first registered.
-  gearbox - The transmission type.
-  powerPS - The power of the car in PS.
-  model - The car model name.
-  kilometer - How many kilometers the car has driven.
-  monthOfRegistration - The month in which which year the car was first registered.
-  fuelType - What type of fuel the car uses.
-  brand - The brand of the car.
-  notRepairedDamage - If the car has a damage which is not yet repaired.
-  dateCreated - The date on which the eBay listing was created.
-  nrOfPictures - The number of pictures in the ad.
-  postalCode - The postal code for the location of the vehicle.
-  lastSeenOnline - When the crawler saw this ad last online.


## The aim of this project is to clean the data and analyze the included used car listings.

In [1]:
import pandas as pd 
import numpy as np 

autos = pd.read_csv('autos.csv', encoding = "Latin-1")
print(autos.isnull().sum())
print("\n")


dateCrawled                0
name                       0
seller                     0
offerType                  0
price                      0
abtest                     0
vehicleType            37869
yearOfRegistration         0
gearbox                20209
powerPS                    0
model                  20484
kilometer                  0
monthOfRegistration        0
fuelType               33386
brand                      0
notRepairedDamage      72060
dateCreated                0
nrOfPictures               0
postalCode                 0
lastSeen                   0
dtype: int64




ate

CEK 123

In [2]:
col_mapping = {"dateCrawled" : "date_crawled", "offerType" : "offer_type", "abtest" : "ab_test", "vehicleType" : "vehicle_type",
              "yearOfRegistration" : "registration_year", "powerPS" : "power_ps", "monthOfRegistration" : "registration_month",
              "fuelType" : "fuel_type", "notRepairedDamage" : "unrepaired_damage", "dateCreated" : "ad_created", "nrOfPictures" : "num_photos",
              "postalCode" : "post_code", "lastSeen" : "last_seen"}

autos = autos.rename(col_mapping, axis = 1)

df = autos

print(df.head)

<bound method NDFrame.head of                date_crawled                                          name  \
0       2016-03-24 11:52:17                                    Golf_3_1.6   
1       2016-03-24 10:58:45                          A5_Sportback_2.7_Tdi   
2       2016-03-14 12:52:21                Jeep_Grand_Cherokee_"Overland"   
3       2016-03-17 16:54:04                            GOLF_4_1_4__3TÜRER   
4       2016-03-31 17:25:20                Skoda_Fabia_1.4_TDI_PD_Classic   
...                     ...                                           ...   
371523  2016-03-14 17:48:27                    Suche_t4___vito_ab_6_sitze   
371524  2016-03-05 19:56:21         Smart_smart_leistungssteigerung_100ps   
371525  2016-03-19 18:57:12            Volkswagen_Multivan_T4_TDI_7DC_UY2   
371526  2016-03-20 19:41:08                        VW_Golf_Kombi_1_9l_TDI   
371527  2016-03-07 19:39:19  BMW_M135i_vollausgestattet_NP_52.720____Euro   

        seller offer_type  price  ab_test veh

## CLEANING DATA

In [3]:
df.describe(include = "all")

date_crawled         name  seller offer_type         price  \
count                371528       371528  371528     371528  3.715280e+05   
unique               280500       233531       2          2           NaN   
top     2016-03-24 14:49:47  Ford_Fiesta  privat    Angebot           NaN   
freq                      7          657  371525     371516           NaN   
mean                    NaN          NaN     NaN        NaN  1.729514e+04   
std                     NaN          NaN     NaN        NaN  3.587954e+06   
min                     NaN          NaN     NaN        NaN  0.000000e+00   
25%                     NaN          NaN     NaN        NaN  1.150000e+03   
50%                     NaN          NaN     NaN        NaN  2.950000e+03   
75%                     NaN          NaN     NaN        NaN  7.200000e+03   
max                     NaN          NaN     NaN        NaN  2.147484e+09   

       ab_test vehicle_type  registration_year  gearbox       power_ps  \
count   371528       333659      371528.000000   351319  371528.000000   
unique       2            8                NaN        2            NaN   
top       test    limousine                NaN  manuell            NaN   
freq    192585        95894                NaN   274214            NaN   
mean       NaN          NaN        2004.577997      NaN     115.549477   
std        NaN          NaN          92.866598      NaN     192.139578   
min        NaN          NaN        1000.000000      NaN       0.000000   
25%        NaN          NaN        1999.000000      NaN      70.000000   
50%        NaN          NaN        2003.000000      NaN     105.000000   
75%        NaN          NaN        2008.000000      NaN     150.000000   
max        NaN          NaN        9999.000000      NaN   20000.000000   

         model      kilometer  registration_month fuel_type       brand  \
count   351044  371528.000000       371528.000000    338142      371528   
unique     251            NaN                 NaN         7          40   
top       golf            NaN                 NaN    benzin  volkswagen   
freq     30070            NaN                 NaN    223857       79640   
mean       NaN  125618.688228            5.734445       NaN         NaN   
std        NaN   40112.337051            3.712412       NaN         NaN   
min        NaN    5000.000000            0.000000       NaN         NaN   
25%        NaN  125000.000000            3.000000       NaN         NaN   
50%        NaN  150000.000000            6.000000       NaN         NaN   
75%        NaN  150000.000000            9.000000       NaN         NaN   
max        NaN  150000.000000           12.000000       NaN         NaN   

       unrepaired_damage           ad_created  num_photos     post_code  \
count             299468               371528    371528.0  371528.00000   
unique                 2                  114         NaN           NaN   
top                 nein  2016-04-03 00:00:00         NaN           NaN   
freq              263182                14450         NaN           NaN   
mean                 NaN                  NaN         0.0   50820.66764   
std                  NaN                  NaN         0.0   25799.08247   
min                  NaN                  NaN         0.0    1067.00000   
25%                  NaN                  NaN         0.0   30459.00000   
50%                  NaN                  NaN         0.0   49610.00000   
75%                  NaN                  NaN         0.0   71546.00000   
max                  NaN                  NaN         0.0   99998.00000   

                  last_seen  
count                371528  
unique               182806  
top     2016-04-07 06:45:59  
freq                     17  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

In [4]:
df["num_photos"].value_counts()

0    371528
Name: num_photos, dtype: int64

In [5]:
df["seller"].value_counts()

privat        371525
gewerblich         3
Name: seller, dtype: int64

In [6]:
df["offer_type"].value_counts()

Angebot    371516
Gesuch         12
Name: offer_type, dtype: int64

-SEMUA NUM PHOTOS NILAINYA 0 , JADI KITA DELETE 
-SELLER TERLALU UMUM PRIVAT DAN GEWERBLICH


In [7]:
col_drop = ["seller","offer_type","num_photos"] #delete column
df2 = df.drop(col_drop, axis = 1)

In [8]:
#print(df.columns)#cek apakah udah ke-delete
#print("\n")
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 20 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   date_crawled        371528 non-null  object
 1   name                371528 non-null  object
 2   seller              371528 non-null  object
 3   offer_type          371528 non-null  object
 4   price               371528 non-null  int64 
 5   ab_test             371528 non-null  object
 6   vehicle_type        333659 non-null  object
 7   registration_year   371528 non-null  int64 
 8   gearbox             351319 non-null  object
 9   power_ps            371528 non-null  int64 
 10  model               351044 non-null  object
 11  kilometer           371528 non-null  int64 
 12  registration_month  371528 non-null  int64 
 13  fuel_type           338142 non-null  object
 14  brand               371528 non-null  object
 15  unrepaired_damage   299468 non-null  object
 16  ad

In [9]:
print(df2.info()) #confirm its deleted

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 17 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   date_crawled        371528 non-null  object
 1   name                371528 non-null  object
 2   price               371528 non-null  int64 
 3   ab_test             371528 non-null  object
 4   vehicle_type        333659 non-null  object
 5   registration_year   371528 non-null  int64 
 6   gearbox             351319 non-null  object
 7   power_ps            371528 non-null  int64 
 8   model               351044 non-null  object
 9   kilometer           371528 non-null  int64 
 10  registration_month  371528 non-null  int64 
 11  fuel_type           338142 non-null  object
 12  brand               371528 non-null  object
 13  unrepaired_damage   299468 non-null  object
 14  ad_created          371528 non-null  object
 15  post_code           371528 non-null  int64 
 16  la

ada kolom bahasa german, terutama di veh type, gearbox, unrepaired_damage, fuel type translate to engilh

In [10]:
print(df2['vehicle_type'].value_counts(dropna = False))
print(df2['gearbox'].value_counts(dropna = False))
print(df2['unrepaired_damage'].value_counts(dropna = False))
print(df2['fuel_type'].value_counts(dropna = False))

limousine     95894
kleinwagen    80023
kombi         67564
NaN           37869
bus           30201
cabrio        22898
coupe         19015
suv           14707
andere         3357
Name: vehicle_type, dtype: int64
manuell      274214
automatik     77105
NaN           20209
Name: gearbox, dtype: int64
nein    263182
NaN      72060
ja       36286
Name: unrepaired_damage, dtype: int64
benzin     223857
diesel     107746
NaN         33386
lpg          5378
cng           571
hybrid        278
andere        208
elektro       104
Name: fuel_type, dtype: int64


In [11]:
print(autos.head())
mapping_fix = {'bus':'bus', 'limousine':'limousine', 'kleinwagen':'supermini', 'kombi':'station_wagon', 'coupe':'coupe',
    'suv':'suv', 'cabrio':'cabrio', 'andere' :'other'}
df2['vehicle_type'] = df2['vehicle_type'].map(mapping_fix, na_action = 'ignore')

mapping_fix = {'manuell' : 'manual', 'automatik' : 'automatic'}
df2['gearbox'] = df2['gearbox'].map(mapping_fix, na_action = 'ignore')

mapping_fix = {'nein' : 'no', 'ja' : 'yes'}
df2['unrepaired_damage'] = df2['unrepaired_damage'].map(mapping_fix, na_action = 'ignore')

mapping_fix = {'benzin' : 'petrol', 'diesel' : 'diesel', 'lpg' : 'lpg', 'cng' : 'cng', 'hybrid' : 'hybrid', 'andere' : 'other',
              'elektro' : 'electric'}
df2['fuel_type'] = df2['fuel_type'].map(mapping_fix, na_action = 'ignore')

print(df2.head())#its fixed yaaa

          date_crawled                            name  seller offer_type  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat    Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat    Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat    Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat    Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat    Angebot   

   price ab_test vehicle_type  registration_year    gearbox  power_ps  model  \
0    480    test          NaN               1993    manuell         0   golf   
1  18300    test        coupe               2011    manuell       190    NaN   
2   9800    test          suv               2004  automatik       163  grand   
3   1500    test   kleinwagen               2001    manuell        75   golf   
4   3600    test   kleinwagen               2008    manuell        69  fabia   

   kilometer  registration_month fuel_type       brand u

##Exploring Kilometer and Price

kta cek angkanya dibuletin gk

In [12]:
print(df2["kilometer"].value_counts())
print("\n")
print(df2['kilometer'].head())

150000    240797
125000     38067
100000     15920
90000      12523
80000      11053
70000       9773
60000       8669
50000       7615
5000        7069
40000       6376
30000       6041
20000       5676
10000       1949
Name: kilometer, dtype: int64


0    150000
1    125000
2    125000
3    150000
4     90000
Name: kilometer, dtype: int64


In [13]:
print(df2["price"].unique().shape)
print("\n")
print(df2["price"].describe())
print("\n")
print(df2["price"].value_counts().head(20))
print("\n")
print(df2["price"].value_counts().head())

(5597,)


count    3.715280e+05
mean     1.729514e+04
std      3.587954e+06
min      0.000000e+00
25%      1.150000e+03
50%      2.950000e+03
75%      7.200000e+03
max      2.147484e+09
Name: price, dtype: float64


0       10778
500      5670
1500     5394
1000     4649
1200     4594
2500     4438
600      3819
3500     3792
800      3784
2000     3432
999      3364
750      3203
650      3150
4500     3053
850      2946
2200     2936
700      2936
1800     2886
900      2874
950      2793
Name: price, dtype: int64


0       10778
500      5670
1500     5394
1000     4649
1200     4594
Name: price, dtype: int64


ada 5597 nilai yang unik pada kolom price. Ada prob pembulatan angka. Ada 10778 mobil dengan harga 0 (which is itu gk make sense), lets say itu ada 53% dari total. Ini mau di remove atau nggak.  Max price 1jt dollar, dimana itu mustahil.

In [14]:
df2["price"].value_counts().sort_index(ascending=False).head(20)

2147483647     1
99999999      15
99000000       1
74185296       1
32545461       1
27322222       1
14000500       1
12345678       9
11111111      10
10010011       1
10000000       8
9999999        3
3895000        1
3890000        1
2995000        1
2795000        1
1600000        2
1300000        1
1250000        2
1234566        1
Name: price, dtype: int64

In [15]:
df2["price"].value_counts().sort_index(ascending=True).head(20)

0     10778
1      1189
2        12
3         8
4         1
5        26
7         3
8         9
9         8
10       84
11        5
12        8
13        7
14        5
15       27
16        2
17        5
18        3
19        3
20       51
Name: price, dtype: int64

ada beberapa listings w/ harga dibawah 20 usd, include 1-100. Ada yg harga 1 juta dollar, sebanyak 15. Karena eBay itu auction, mungkin bakal di simpan, namun dihapus diatasnya, sejak harga lompat ke nilai yg tidak realistis.

In [16]:
print(((df2["price"]>0)&(df2["price"]<350000)).value_counts())
print("\n")
print(((df2["price"]>=0).value_counts()))

True     360631
False     10897
Name: price, dtype: int64


True    371528
Name: price, dtype: int64


In [17]:
price_filter = (df2["price"]>0)&(df2["price"]<350000) #kta fiter harganya harus bernilai positif dan price nya dibawah 350k
df3 = df2
df3 = df2[price_filter]
df3["price"].describe()

count    360631.000000
mean       5894.855298
std        8792.034481
min           1.000000
25%        1250.000000
50%        3000.000000
75%        7490.000000
max      349000.000000
Name: price, dtype: float64

In [18]:
print(((df3["price"]>0)&(df3["price"]<350000)).value_counts())
print("\n")
print(((df3["price"]>=-1).value_counts()))

True    360631
Name: price, dtype: int64


True    360631
Name: price, dtype: int64


## Exploring the Date Column

kolom dengan format tanggal (date)

-  date_crawled
-  registration_month
-  reistration_year
-  ad_created
-  last_seen 

berikt adalah kombinasi dari dates yng crawled dan meta informasi yang crawled. Non regis dates di simpan sebagai strings

terus kita explore tiap kolomnya biar tahu listingnya

In [19]:
df3[["date_crawled","ad_created","last_seen"]][0:5]

date_crawled           ad_created            last_seen
0  2016-03-24 11:52:17  2016-03-24 00:00:00  2016-04-07 03:16:57
1  2016-03-24 10:58:45  2016-03-24 00:00:00  2016-04-07 01:46:50
2  2016-03-14 12:52:21  2016-03-14 00:00:00  2016-04-05 12:47:46
3  2016-03-17 16:54:04  2016-03-17 00:00:00  2016-03-17 17:40:17
4  2016-03-31 17:25:20  2016-03-31 00:00:00  2016-04-06 10:17:21

## Review about matrix, and how it works

In [20]:
A = [[1, 4, 5, 12], 
    [-5, 8, 9, 0],
    [-6, 7, 11, 19]]

print("A =", A) 
print("A[1] =", A[1])      # 2nd row
print("A[1][2] =", A[1][2])   # 3rd element of 2nd row
print("A[0][-1] =", A[0][-1])   # Last element of 1st Row

column = [];        # empty list
for row in A:
  column.append(row[2])   

print("3rd column =", column)
print("\n")
print(A)

A = [[1, 4, 5, 12], [-5, 8, 9, 0], [-6, 7, 11, 19]]
A[1] = [-5, 8, 9, 0]
A[1][2] = 9
A[0][-1] = 12
3rd column = [5, 9, 11]


[[1, 4, 5, 12], [-5, 8, 9, 0], [-6, 7, 11, 19]]


In [21]:
from numpy import *

In [22]:

m_ar = array([['Mon',18,20,22,17],['Tue',11,18,21,18],
   ['Wed',15,21,20,19],['Thu',11,20,22,21],
   ['Fri',18,17,23,22],['Sat',12,22,20,18],
   ['Sun',13,15,19,16]])
m_r = append(m_ar,[['Avg',12,15,13,11]],0)

print(m_ar)
print("\n")
print(m_r)

[['Mon' '18' '20' '22' '17']
 ['Tue' '11' '18' '21' '18']
 ['Wed' '15' '21' '20' '19']
 ['Thu' '11' '20' '22' '21']
 ['Fri' '18' '17' '23' '22']
 ['Sat' '12' '22' '20' '18']
 ['Sun' '13' '15' '19' '16']]


[['Mon' '18' '20' '22' '17']
 ['Tue' '11' '18' '21' '18']
 ['Wed' '15' '21' '20' '19']
 ['Thu' '11' '20' '22' '21']
 ['Fri' '18' '17' '23' '22']
 ['Sat' '12' '22' '20' '18']
 ['Sun' '13' '15' '19' '16']
 ['Avg' '12' '15' '13' '11']]


## end of review

In [23]:
ab = ["date_crawled","ad_created","last_seen"]
print(ab[1])

ad_created


In [24]:
(df3[ab[0]]
 .str[0:10]
 .value_counts(normalize=True, dropna = False) #normalize = true biar unique values di itung lagi misal list = 3,1,2,3. Peluang 3 = 2/4
 .sort_index() #biar di sort based on labelnya, bisa diliat datesnya berurutan
)

2016-03-05    0.025547
2016-03-06    0.014483
2016-03-07    0.035657
2016-03-08    0.033469
2016-03-09    0.034115
2016-03-10    0.032646
2016-03-11    0.032773
2016-03-12    0.036242
2016-03-13    0.015783
2016-03-14    0.036328
2016-03-15    0.033425
2016-03-16    0.030205
2016-03-17    0.031647
2016-03-18    0.013119
2016-03-19    0.035272
2016-03-20    0.036400
2016-03-21    0.035679
2016-03-22    0.032493
2016-03-23    0.032002
2016-03-24    0.029914
2016-03-25    0.032801
2016-03-26    0.031975
2016-03-27    0.030225
2016-03-28    0.035064
2016-03-29    0.034126
2016-03-30    0.033536
2016-03-31    0.031872
2016-04-01    0.034146
2016-04-02    0.035094
2016-04-03    0.038813
2016-04-04    0.037626
2016-04-05    0.012780
2016-04-06    0.003128
2016-04-07    0.001617
Name: date_crawled, dtype: float64

In [25]:
(df3[ab[1]]
 .str[0:10]
 .value_counts(normalize=True, dropna = False) #normalize = true biar unique values di itung lagi misal list = 3,1,2,3. Peluang 3 = 2/4
 .sort_index() #biar di sort based on labelnya, bisa diliat datesnya berurutan
)

2014-03-10    0.000003
2015-03-20    0.000003
2015-06-11    0.000003
2015-06-18    0.000003
2015-08-07    0.000003
                ...   
2016-04-03    0.039001
2016-04-04    0.037734
2016-04-05    0.011613
2016-04-06    0.003120
2016-04-07    0.001553
Name: ad_created, Length: 114, dtype: float64

In [26]:
(df3[ab[2]]
 .str[0:10]
 .value_counts(normalize=True, dropna = False) #normalize = true biar unique values di itung lagi misal list = 3,1,2,3. Peluang 3 = 2/4
 .sort_index() #biar di sort based on labelnya, bisa diliat datesnya berurutan
)

2016-03-05    0.001264
2016-03-06    0.004098
2016-03-07    0.005202
2016-03-08    0.007939
2016-03-09    0.009824
2016-03-10    0.011460
2016-03-11    0.012955
2016-03-12    0.023240
2016-03-13    0.008410
2016-03-14    0.012176
2016-03-15    0.016321
2016-03-16    0.016418
2016-03-17    0.028700
2016-03-18    0.006888
2016-03-19    0.016330
2016-03-20    0.019885
2016-03-21    0.020026
2016-03-22    0.020506
2016-03-23    0.018016
2016-03-24    0.019164
2016-03-25    0.019000
2016-03-26    0.015958
2016-03-27    0.016718
2016-03-28    0.022189
2016-03-29    0.023284
2016-03-30    0.023725
2016-03-31    0.024244
2016-04-01    0.023897
2016-04-02    0.024967
2016-04-03    0.025308
2016-04-04    0.025533
2016-04-05    0.126964
2016-04-06    0.218952
2016-04-07    0.130438
Name: last_seen, dtype: float64

crawler record tanggal, dimana membuat bisa dicari hari apa yg listing ilangin, yg kyknya mobil nya dah kejual

3 hari terakhir terdiri dari ketidak proprosian dari jumlah "last-seen" value. Diberitahu bahwa 6-10x value dari haris ebelumnya, kyknya ada spikes dan nilainya kyknya crawling priod dan tidak mengindikasi car sales.

## Incorrect registration year data

mobil gk bisa regist sebelum listing, mobil manapun dengan regist diatas 2016 biasanya tak akurat

satu opsi = remove listings dengan valuenya. Cek presentase dari data kita yang tidak valid dalam kolom:

In [27]:
filter123 = (df3["registration_year"]>1900)&(df3["registration_year"]<2017)
print(filter123)
correct = df3[filter123].shape[0]
print("\n")
print("correct = ",correct)
correct_prcnt = correct/df3.shape[0]
print("correct %= ",1 - correct_prcnt)

0         True
1         True
2         True
3         True
4         True
          ... 
371523    True
371524    True
371525    True
371526    True
371527    True
Name: registration_year, Length: 360631, dtype: bool


correct =  346656
correct %=  0.03875152163846152


## Explor harga by Brandnya

-  kita lihat disini apakah harga sesuai dengan kualitas brandnya

In [28]:
# kita bikin dulu data yg udah

clean_df = df3[filter123]
clean_df["registration_year"].value_counts(normalize = True).head(10)

2000    0.066700
1999    0.063553
2005    0.062670
2006    0.057709
2001    0.056956
2003    0.056558
2004    0.056174
2002    0.054290
2007    0.050500
1998    0.049692
Name: registration_year, dtype: float64

-  terlihat bahwa mostly mobil yg terjual teregister dalam 20 thn terakhir

In [29]:
brand_cnt = clean_df["brand"].value_counts(normalize=True)
brand_cnt

volkswagen        0.211703
bmw               0.109873
opel              0.106411
mercedes_benz     0.096840
audi              0.089544
ford              0.068918
renault           0.047517
peugeot           0.030154
fiat              0.025691
seat              0.018661
skoda             0.015687
mazda             0.015384
smart             0.014331
citroen           0.013950
nissan            0.013598
toyota            0.012932
hyundai           0.009972
sonstige_autos    0.009491
mini              0.009384
volvo             0.009147
mitsubishi        0.008236
honda             0.007532
kia               0.006915
suzuki            0.006364
alfa_romeo        0.006309
porsche           0.006205
chevrolet         0.005022
chrysler          0.003863
dacia             0.002495
jeep              0.002192
land_rover        0.002166
daihatsu          0.002161
subaru            0.002117
jaguar            0.001734
saab              0.001465
daewoo            0.001457
trabant           0.001408
l


prshaan german mewakili 4 dari 5 top five brands. hampir 50% dari overall. Volkswagen terpopuler, bahkan dia setara jumlah dari rank 2 dan 3

banyak brand yg gk punya nilai presentase yg signifikan. Maka kita limit brand untuk di analisis yg mewakili 5% dari keseluruhan listing brand

In [30]:
filter345 = brand_cnt > 0.05

common_brand = brand_cnt[filter345]

common_brand

volkswagen       0.211703
bmw              0.109873
opel             0.106411
mercedes_benz    0.096840
audi             0.089544
ford             0.068918
Name: brand, dtype: float64

In [31]:
common_brand.index

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford'], dtype='object')

In [32]:
common_brand_index = common_brand.index

mean_price_per_brand = {}

for brand in common_brand_index:
    filter789 = clean_df["brand"] == brand
    cars = clean_df[filter789]
    brand_mean = cars["price"].mean()
    mean_price_per_brand[brand] = int(brand_mean)

mean_price_per_brand

{'volkswagen': 5400,
 'bmw': 8449,
 'opel': 2971,
 'mercedes_benz': 8541,
 'audi': 9086,
 'ford': 3696}

In [33]:
filter789.value_counts()

False    322765
True      23891
Name: brand, dtype: int64

dari top 5 brand, ada perbedaaan harga:

-  audi, bmw, merci termahal
-  ford and opel top murah
-  volkswagen diantara mahal dan murah (so so),= alasan volkswag populer

## Exploring Mileage (Jarak tempuh)

In [34]:
import pandas as pd

In [39]:
mean_mile_per_brand = {}

for brand in common_brand_index:
    fltr = clean_df["brand"] == brand
    cars = clean_df[fltr]
    mile_mean = cars["kilometer"].mean()
    mean_mile_per_brand[brand] = int(mile_mean)

mean_mile = pd.Series(mean_mile_per_brand).sort_values(ascending=False)
print(mean_mile)
print()
mean_prices = pd.Series(mean_price_per_brand).sort_values(ascending=False)
print(mean_prices)   

bmw              132800
mercedes_benz    130573
audi             129443
opel             128722
volkswagen       128386
ford             123662
dtype: int64

audi             9086
mercedes_benz    8541
bmw              8449
volkswagen       5400
ford             3696
opel             2971
dtype: int64


In [46]:
brand_info = pd.DataFrame(columns = ["mean_mile", "mean_prices"])
brand_info

Empty DataFrame
Columns: [mean_mile, mean_prices]
Index: []

In [47]:
brand_info["mean_prices"] = mean_prices
brand_info["mean_mile"] = mean_mile

brand_info.sort_values(by = "mean_prices")

mean_mile  mean_prices
opel              128722         2971
ford              123662         3696
volkswagen        128386         5400
bmw               132800         8449
mercedes_benz     130573         8541
audi              129443         9086

Range dari mile si mobil tidak begitu varian seiring dengan harga si brandnya. Namun, jatuh hasilnya pada top 10% brand. Ada sedikit trend pada mobil yang mahal memiliki mile yg lebih besar, dengan harga mobil yg sedikit lbh mahal dengan mile kecil.

## Kesimpulan

-  semkin tinggin harga tidak begitu menggambarkan performa mile nya, contohnya brand opel dengan 2971 mean prices dan 128722 kilo mile.

-  bandingkan dengan brand yg tidak begitu luxurious (ford dan volkswagen) dengan nilai rata-rata 3696 dan 5400. Keduanya memiliki mile yang tidak jauh beda dengam opel.